drivers/filesystems/fastfat/fsctl.c

```diff
@@ -1125,6 +1137,11 @@ VfatLockOrUnlockVolume(
         return STATUS_ACCESS_DENIED;
     }
 
+    if (Lock)
+    {
+        FsRtlNotifyVolumeEvent(IrpContext->Stack->FileObject, FSRTL_VOLUME_LOCK);
```

Comment:
Given where you notify unlocking, this notification is way too early.

Reply:
fastfat_new does it also "early".